src/code/z_parameter.c

```diff
@@ -52,38 +332,750 @@
+        }
+        return ITEM_NONE;
+
+    } else if (item == ITEM_BOMBCHU) {
+        if (INV_CONTENT(ITEM_BOMBCHU) != ITEM_BOMBCHU) {
+            INV_CONTENT(ITEM_BOMBCHU) = ITEM_BOMBCHU;
+            AMMO(ITEM_BOMBCHU) = 10;
+            return ITEM_NONE;
+        }
+        if ((AMMO(ITEM_BOMBCHU) += 10) > CUR_CAPACITY(UPG_BOMB_BAG)) {
+            AMMO(ITEM_BOMBCHU) = CUR_CAPACITY(UPG_BOMB_BAG);
+        }
+        return ITEM_NONE;
+
+    } else if ((item >= ITEM_BOMBCHUS_20) && (item <= ITEM_BOMBCHUS_5)) {
+        if (gSaveContext.save.inventory.items[SLOT_BOMBCHU] != ITEM_BOMBCHU) {
+            INV_CONTENT(ITEM_BOMBCHU) = ITEM_BOMBCHU;
+            AMMO(ITEM_BOMBCHU) += sAmmoRefillCounts[item - ITEM_BOMBCHUS_20 + 8];
+
+            if (AMMO(ITEM_BOMBCHU) > CUR_CAPACITY(UPG_BOMB_BAG)) {
+                AMMO(ITEM_BOMBCHU) = CUR_CAPACITY(UPG_BOMB_BAG);
+            }
+            return ITEM_NONE;
+        }
+
+        if ((AMMO(ITEM_BOMBCHU) += sAmmoRefillCounts[item - ITEM_BOMBCHUS_20 + 8]) > CUR_CAPACITY(UPG_BOMB_BAG)) {
+            AMMO(ITEM_BOMBCHU) = CUR_CAPACITY(UPG_BOMB_BAG);
+        }
+        return ITEM_NONE;
+
+    } else if ((item >= ITEM_ARROWS_10) && (item <= ITEM_ARROWS_50)) {
+        AMMO(ITEM_BOW) += sAmmoRefillCounts[item - ITEM_ARROWS_10 + 4];
+
+        if ((AMMO(ITEM_BOW) >= CUR_CAPACITY(UPG_QUIVER)) || (AMMO(ITEM_BOW) < 0)) {
+            AMMO(ITEM_BOW) = CUR_CAPACITY(UPG_QUIVER);
+        }
+        return ITEM_BOW;
+
+    } else if (item == ITEM_OCARINA) {
+        INV_CONTENT(ITEM_OCARINA) = 0;
+        return ITEM_NONE;
+
+    } else if (item == ITEM_BEAN) {
+        if (INV_CONTENT(ITEM_BEAN) == ITEM_NONE) {
+            INV_CONTENT(item) = item;
+            AMMO(ITEM_BEAN) = 1;
+        } else if (AMMO(ITEM_BEAN) < 20) {
+            AMMO(ITEM_BEAN)++;
+        } else {
+            AMMO(ITEM_BEAN) = 20;
+        }
+        return ITEM_NONE;
+
+    } else if ((item >= ITEM_REMAINS_ODOLWA) && (item <= ITEM_REMAINS_TWINMOLD)) {
+        SET_QUEST_ITEM(item - ITEM_REMAINS_ODOLWA + QUEST_REMAINS_ODOWLA);
+        return ITEM_NONE;
+
+    } else if (item == ITEM_HEART) {
+        Health_ChangeBy(globalCtx, 0x10);
+        return item;
+
+    } else if (item == ITEM_MAGIC_SMALL) {
+        Interface_AddMagic(globalCtx, 0x18);
+        if (!(gSaveContext.save.weekEventReg[12] & 0x80)) {
+            gSaveContext.save.weekEventReg[12] |= 0x80;
+            return ITEM_NONE;
+        }
+        return item;
+
+    } else if (item == ITEM_MAGIC_LARGE) {
+        Interface_AddMagic(globalCtx, 0x30);
+        if (!(gSaveContext.save.weekEventReg[12] & 0x80)) {
+            gSaveContext.save.weekEventReg[12] |= 0x80;
+            return ITEM_NONE;
+        }
+        return item;
+
+    } else if ((item >= ITEM_RUPEE_GREEN) && (item <= ITEM_RUPEE_HUGE)) {
+        Rupees_ChangeBy(sAmmoRefillCounts[item - ITEM_RUPEE_GREEN + 12]); // TODO: Different to OoT? Or wrong.
```

Comment:
What's the significance of the 12?

Reply:
I don't know, hence the comment. Something about this is wrong but I haven't worked out what yet

Reply:
Alright, I went through this and found something much more logical. `sAmmoRefillCounts` should really be 4 separate pieces of data, and the +12 is just indexing where the real data should be starting from (This'll have to be fixed in OoT as well)